apps/src/redux/scriptSelectionRedux.js

```diff
@@ -0,0 +1,129 @@
+import { PropTypes } from 'react';
+
+// Reducer for script selection in teacher dashboard.
+// Tab specific reducers can import actions from this file
+// if they need to respond to a script changing.
+
+const DEFAULT_SCRIPT_NAME = "Express Course";
+
+// Action type constants
+export const SET_SCRIPT = 'scriptSelection/SET_SCRIPT';
+export const SET_VALID_SCRIPTS = 'scriptSelection/SET_VALID_SCRIPTS';
+
+// Action creators
+export const setScriptId = scriptId => ({ type: SET_SCRIPT, scriptId});
+export const setValidScripts = (validScripts, studentScriptIds, validCourses, assignedCourseId) => (
+  {type: SET_VALID_SCRIPTS, validScripts, studentScriptIds, validCourses, assignedCourseId}
+);
+
+/**
+ * Shape for a validScript
+ */
+export const validScriptPropType = PropTypes.shape({
+  category: PropTypes.string.isRequired,
+  category_priority: PropTypes.number.isRequired,
+  id: PropTypes.number.isRequired,
+  name: PropTypes.string.isRequired,
+  position: PropTypes.number,
+});
+
+// Initial state of scriptSelectionRedux
+const initialState = {
+  scriptId: null,
+  validScripts: [],
+};
+
+export default function scriptSelection(state=initialState, action) {
+  if (action.type === SET_SCRIPT) {
+    return {
+      ...state,
+      scriptId: action.scriptId,
+    };
+  }
+
+  if (action.type === SET_VALID_SCRIPTS) {
```

Comment:
From @davidsbailey on #22289: this is starting to get a bit confusing, because now the part of the function that determines the scriptId is also modifying validScripts. As a first step, I would recommend moving the logic for if (validScripts.length === 0) {validScripts.push(defaultScript)} to line 226 and replacing the default and last case statement here with just

default:
  scriptId = validScripts[0].id
and also adding a comment explaining that one section computes validScripts and the other computes scriptId.

Ideally you would extract 1 or 2 self-documenting functions like this:

if (action.type === SET_VALID_SCRIPTS) {
  const validScripts = computeValidScripts(...)
  const scriptId = getDefaultScriptId(...)
}
__________________
Can save for a follow up, just want to make sure we're not losing this thought and I don't know where exactly it should be tracked.

Reply:
to be clear, the part of the above comment that's still relevant is just this part:

> Ideally you would extract 1 or 2 self-documenting functions like this:
>if (action.type === SET_VALID_SCRIPTS) {
>  const validScripts = computeValidScripts(...)
>  const scriptId = getDefaultScriptId(...)
>}

Reply:
I think that's a great idea, but the goal of this PR is to move the existing code into a shared space, and I think this is something we could still do after this PR is merged.